Automattic/jetpack
# Add tools to find and eliminate jQuery usage

Fixes #16409

The idea here is to show where we use jQuery and gradually eliminate it.

It relies in part on a small eslint plugin I wrote called `no-jquery` (https://github.com/Automattic/eslint-plugin-no-jquery), which detects and warns about all jQuery usage, as well as a commonly-used jQuery eslint plugin that recommends native-JS alternatives.

I did not make this part of the default eslint rules because it will add a bunch of noise. The idea is to merge this and use it as the basis for future changes on a file-by-file basis.

Limitations:
* Catches variables called jQuery, or any variables that start with `$`. This is bother overly broad and overly narrow - users can alias jQuery to vars other than $ and this won't pick it up. 
* Detection of `$` as a _prefix_ is supposed to catch things like `var $mything = jQuery.find( '.mything' ); $mything.find( '.childthing');`, but this depends on the convention of prefixing jQuery objects with `$`

#### Changes proposed in this Pull Request:

* Add npm/yarn command to lint jquery usage

#### Testing instructions:
* Run `yarn run lint-jquery`
* You should see a list of details
* Run `yarn run lint-jquery-files-only`
* You should see just a list of relative filenames
* Run `yarn run lint-jquery-rules-only`
* You should see a list of rules with violation counts (helps with figuring out which rules to automate fixers for first)
